File: shuttle-api/src/ShuttleApi.Domain/Trips/TripInspectionItem.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Trips;

public enum InspectionCategory
{
    ExteriorMechanical,
    SafetyEquipment,
    InteriorComfort,
    CommunicationsNavigation
}

public sealed class TripInspectionItem : Entity<Guid>
{
    public Guid PreInspectionId { get; private set; }
    public string ItemName { get; private set; } = string.Empty;
    public InspectionCategory Category { get; private set; }
    public bool Passed { get; private set; }
    public string? Notes { get; private set; }

    private TripInspectionItem() { }

    public static TripInspectionItem Create(
        Guid preInspectionId,
        string itemName,
        InspectionCategory category,
        bool passed,
        string? notes)
    {
        return new TripInspectionItem
        {
            Id = Guid.NewGuid(),
            PreInspectionId = preInspectionId,
            ItemName = itemName,
            Category = category,
            Passed = passed,
            Notes = notes
        };
    }
}

File: shuttle_tablet/lib/features/trips/domain/entities/trip_inspection_item.dart
import 'package:equatable/equatable.dart';

enum InspectionCategory {
  exteriorMechanical,
  safetyEquipment,
  interiorComfort,
  communicationsNavigation,
}

class TripInspectionItem extends Equatable {
  final String id;
  final String preInspectionId;
  final String itemName;
  final InspectionCategory category;
  final bool passed;
  final String? notes;

  const TripInspectionItem({
    required this.id,
    required this.preInspectionId,
    required this.itemName,
    required this.category,
    required this.passed,
    this.notes,
  });

  @override
  List<Object?> get props =>
      [id, preInspectionId, itemName, category, passed, notes];
}
